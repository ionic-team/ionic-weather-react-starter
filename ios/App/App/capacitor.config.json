{
  "appId": "io.ionic.starter",
  "appName": "ionic-weather-react-starter",
  "bundledWebRuntime": false,
  "npmClient": "npm",
  "webDir": "build",
  "cordova": {}
}
