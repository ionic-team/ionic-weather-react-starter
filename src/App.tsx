import React from 'react';
import { Redirect, Route } from 'react-router-dom';
import { IonApp, IonIcon, IonLabel, IonRouterOutlet, IonTabBar, IonTabButton, IonTabs } from '@ionic/react';
import { IonReactRouter } from '@ionic/react-router';
import { cloud, calendar, sunny } from 'ionicons/icons';
import CurrentWeatherPage from './pages/CurrentWeather';
import ForecastPage from './pages/Forecast';
import Tab3 from './pages/UVIndex';

/* Core CSS required for Ionic components to work properly */
import '@ionic/react/css/core.css';

/* Basic CSS for apps built with Ionic */
import '@ionic/react/css/normalize.css';
import '@ionic/react/css/structure.css';
import '@ionic/react/css/typography.css';

/* Optional CSS utils that can be commented out */
import '@ionic/react/css/padding.css';
import '@ionic/react/css/float-elements.css';
import '@ionic/react/css/text-alignment.css';
import '@ionic/react/css/text-transformation.css';
import '@ionic/react/css/flex-utils.css';
import '@ionic/react/css/display.css';

/* Theme variables */
import './theme/variables.css';
import './theme/styles.css';

const App: React.FC = () => (
  <IonApp>
    <IonReactRouter>
      <IonTabs>
        <IonRouterOutlet>
          <Route path="/current-weather" component={CurrentWeatherPage} exact={true} />
          <Route path="/forecast" component={ForecastPage} exact={true} />
          <Route path="/uv-index" component={Tab3} />
          <Route path="/" render={() => <Redirect to="/current-weather" />} exact={true} />
        </IonRouterOutlet>
        <IonTabBar slot="bottom" color="primary">
          <IonTabButton tab="tab1" href="/current-weather">
            <IonIcon icon={cloud} />
            <IonLabel>Current Weather</IonLabel>
          </IonTabButton>
          <IonTabButton tab="tab2" href="/forecast">
            <IonIcon icon={calendar} />
            <IonLabel>Forecast</IonLabel>
          </IonTabButton>
          <IonTabButton tab="tab3" href="/uv-index">
            <IonIcon icon={sunny} />
            <IonLabel>UV Index</IonLabel>
          </IonTabButton>
        </IonTabBar>
      </IonTabs>
    </IonReactRouter>
  </IonApp>
);

export default App;
