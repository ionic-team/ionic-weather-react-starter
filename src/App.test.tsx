import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

it('renders without crashing', () => {
  const { container } = render(<App />);
  expect(container.innerHTML).toBeTruthy();
});

it.each([
  [0, "Tab One"],
  [1, "Tab Two"],
  [2, "Tab Three"]
])("contains the proper text for tab %i", (tab, text) => {
  const { container } = render(<App />);
  expect(
    container.querySelectorAll("ion-tab-button")[tab as number].textContent
  ).toEqual(text);
});

it('renders consistently', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
