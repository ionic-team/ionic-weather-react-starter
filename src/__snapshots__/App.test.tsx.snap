// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders consistently 1`] = `
<DocumentFragment>
  <ion-app>
    <div
      style="display: flex; position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; flex-direction: column; width: 100%; height: 100%; contain: layout size style;"
    >
      <div
        class="tabs-inner"
        style="position: relative; flex: 1; contain: layout size style;"
      >
        <ion-router-outlet>
          <div
            class="ion-page"
            style="z-index: 101;"
          >
            <ion-header>
              <ion-toolbar>
                <ion-title>
                  Current Weather
                </ion-title>
              </ion-toolbar>
            </ion-header>
            <ion-content
              class="ion-text-center ion-padding"
            >
              <div
                class="information"
              >
                <kws-temperature
                  class="primary-value"
                  scale="F"
                  temperature="302"
                />
              </div>
              <kws-condition
                condition="200"
              />
            </ion-content>
          </div>
        </ion-router-outlet>
      </div>
      <ion-tab-bar
        color="primary"
        current-path="/current-weather"
        selected-tab="tab1"
        slot="bottom"
      >
        <ion-tab-button
          href="/current-weather"
          tab="tab1"
        >
          <ion-icon />
          <ion-label>
            Current Weather
          </ion-label>
        </ion-tab-button>
        <ion-tab-button
          href="/forecast"
          tab="tab2"
        >
          <ion-icon />
          <ion-label>
            Forecast
          </ion-label>
        </ion-tab-button>
        <ion-tab-button
          href="/uv-index"
          tab="tab3"
        >
          <ion-icon />
          <ion-label>
            UV Index
          </ion-label>
        </ion-tab-button>
      </ion-tab-bar>
    </div>
  </ion-app>
</DocumentFragment>
`;
